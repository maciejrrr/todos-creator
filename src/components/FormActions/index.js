import React from 'react';
import { Icon } from 'antd';
import styled, { css } from 'styled-components';

import { metrics, colors } from '../../theme';

const ActionsContainer = styled.div`
  margin-top: ${props => (props.inline ? 0 : metrics.smallMargin)}px;
  margin-left: ${metrics.smallMargin}px;
  display: flex;
  align-items: center;
`;

const SubmitButton = styled.button`
  border: 0;
  border-radius: ${metrics.borderRadius}px;
  background-image: linear-gradient(${colors.gradient});
  padding: ${metrics.smallMargin}px ${metrics.doubleMargin}px;
  color: ${colors.white};
  ${props =>
    props.disabled &&
    css`
      cursor: not-allowed;
      background-image: none;
      background-color: ${colors.white};
      color: ${colors.grayText};
    `};
`;

const CancelForm = styled(Icon)`
  cursor: pointer;
  font-size: ${metrics.fontSize.xl}px;
  margin-left: ${metrics.smallMargin}px;
  color: ${colors.grayText};
`;

const FormActions = ({ focused, disabled, text, resetForm, inline }) => {
  let content = null;
  if (focused) {
    content = (
      <ActionsContainer inline={inline}>
        <SubmitButton htmlType="submit" disabled={disabled}>
          {text}
        </SubmitButton>
        <CancelForm type="close" onClick={resetForm} />
      </ActionsContainer>
    );
  }
  return content;
};

export default FormActions;
