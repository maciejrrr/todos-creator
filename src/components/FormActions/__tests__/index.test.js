import FormActions from '../index';

describe('FormActions component', () => {
  let props;
  beforeEach(() => {
    props = {
      focused: false,
      disabled: false,
      text: 'test',
      resetForm: jest.fn(),
    };
  });

  describe('snapshots', () => {
    it('returns null', () => {
      const tree = renderer.create(<FormActions {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders form actions', () => {
      const tree = renderer
        .create(<FormActions {...props} focused={true} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders form actions with disable submit button', () => {
      const tree = renderer
        .create(<FormActions {...props} focused={true} disabled={true} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
