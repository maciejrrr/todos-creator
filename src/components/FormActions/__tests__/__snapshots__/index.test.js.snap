// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormActions component snapshots renders form actions 1`] = `
<div
  className="sc-bdVaJa jIKszl"
>
  <button
    className="sc-bwzfXH PLgtL"
    disabled={false}
  >
    test
  </button>
  <i
    className="anticon anticon-close sc-htpNat oPmsX"
    onClick={[Function]}
  />
</div>
`;

exports[`FormActions component snapshots renders form actions with disable submit button 1`] = `
<div
  className="sc-bdVaJa jIKszl"
>
  <button
    className="sc-bwzfXH hGZcQI"
    disabled={true}
  >
    test
  </button>
  <i
    className="anticon anticon-close sc-htpNat oPmsX"
    onClick={[Function]}
  />
</div>
`;

exports[`FormActions component snapshots returns null 1`] = `null`;
